fluentd/logger/supervisor.rb

```diff
@@ -0,0 +1,582 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'etc'
+require 'fcntl'
+require 'logger'
+
+begin
+  require 'sigdump/setup'
+rescue Exception
+  # ignore LoadError and others (related with signals): it may raise these errors in Windows
+end
+
+require 'fluent/config'
+require 'fluent/env'
+require 'fluent/engine'
+require 'fluent/log'
+require 'fluent/plugin'
+require 'fluent/rpc'
+require 'fluent/system_config'
+
+module Fluent
+  class Supervisor
+    def self.get_etc_passwd(user)
+      if user.to_i.to_s == user
+        Etc.getpwuid(user.to_i)
+      else
+        Etc.getpwnam(user)
+      end
+    end
+
+    def self.get_etc_group(group)
+      if group.to_i.to_s == group
+        Etc.getgrgid(group.to_i)
+      else
+        Etc.getgrnam(group)
+      end
+    end
+
+    class LoggerInitializer
+      def initialize(path, level, chuser, chgroup, opts)
+        envPath = ENV["LOGGING_FILE_PATH"]
+        $openShiftLogger = nil
+        if !envPath.nil?
```

Comment:
This if block wont be needed if we default.  for future ref, this is also valid `unless envPath.nil?`